src/Host/Broker/Impl/About/AboutController.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Management;
+using System.Reflection;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.R.Host.Broker.Interpreters;
+using Microsoft.R.Host.Broker.Security;
+using Microsoft.R.Host.Protocol;
+using Microsoft.R.Interpreters;
+
+namespace Microsoft.R.Host.Broker.About {
+    [Authorize(Policy = Policies.RUser)]
+    [Route("/about")]
+    public class AboutController : Controller {
+        private readonly InterpreterManager _interpManager;
+
+        public AboutController(InterpreterManager interpManager) {
+            _interpManager = interpManager;
+        }
+
+        [HttpGet]
+        public AboutHost Get() {
+            var a = new AboutHost();
+
+            a.Version = Assembly.GetExecutingAssembly().GetName().Version;
+            a.OS = Environment.OSVersion;
+            a.Is64BitOperatingSystem = Environment.Is64BitOperatingSystem;
+            a.Is64BitProcess = Environment.Is64BitProcess;
+            a.ProcessorCount = Environment.ProcessorCount;
+            a.WorkingSet = Environment.WorkingSet;
+
+            var query = new SelectQuery(@"Select * from Win32_OperatingSystem");
+            using (var search = new ManagementObjectSearcher(query)) {
+                foreach (var mo in search.Get()) {
+                    a.TotalVirtualMemory = GetValue(mo, "TotalVirtualMemorySize");
+                    a.FreeVirtualMemory = GetValue(mo, "FreeVirtualMemory");
+                    a.TotalPhysicalMemory = GetValue(mo, "TotalVisibleMemorySize");
+                    a.FreePhysicalMemory = GetValue(mo, "FreePhysicalMemory");
+                    break;
+                }
+            }
+
+            a.Interpreters = new RInstallation().GetCompatibleEngines().Select(e => e.Name).ToArray();
```

Comment:
I think that was back when they only had IDs. Those were user-specified for custom interpreters, and empty string for the one and only auto-detected one.

With all the recent changes, interpreters have both IDs (again, user-specified for custom, and GUIDs for auto-detected) and names (user-specified for custom, obtained from `RInstallation` for auto-detected). That shows up in JSON returned from `/interpreters`, e.g. here's mine:
```json
[
  {
    "id": "c2a5ee84-c6af-4830-8630-475fe0f59e5f",
    "name": "R 3.2.1",
    "path": "C:\\Program Files\\R\\R-3.2.1",
    "binPath": "C:\\Program Files\\R\\R-3.2.1\\bin\\x64",
    "version": "3.2.1"
  },
  {
    "id": "4e275ceb-3c90-4262-83c2-d94bd6d57644",
    "name": "R 3.2.2",
    "path": "C:\\Program Files\\R\\R-3.2.2",
    "binPath": "C:\\Program Files\\R\\R-3.2.2\\bin\\x64",
    "version": "3.2.2"
  },
  {
    "id": "bed80ba6-674d-4aaa-bedd-a7c1a73749f4",
    "name": "R 3.2.3",
    "path": "C:\\Program Files\\Microsoft\\MRO\\R-3.2.3",
    "binPath": "C:\\Program Files\\Microsoft\\MRO\\R-3.2.3\\bin\\x64",
    "version": "3.2.3"
  },
  {
    "id": "c568f648-b68d-46e5-9952-c882ca31bb48",
    "name": "R 3.2.5",
    "path": "C:\\Program Files\\R\\R-3.2.5",
    "binPath": "C:\\Program Files\\R\\R-3.2.5\\bin\\x64",
    "version": "3.2.5"
  }
]
```